dsl/jpql/src/main/kotlin/com/linecorp/kotlinjdsl/dsl/jpql/Jpql.kt

```diff
@@ -568,6 +568,56 @@ open class Jpql : JpqlDsl {
         return Expressions.ceiling(value.toExpression())
     }
 
+    /**
+     * Creates an expression that represents the mod of values.
+     * The values are each enclosed in parentheses.
+     *
+     * This is the same as ```(value1) % (value2)```.
+     */
+    @JvmName("modWithParentheses")
+    @SinceJdsl("3.3.2")
+    fun <T : Number, S : T?> mod(value1: Expressionable<@Exact T>, value2: S): Expression<T> {
+        return Expressions.mod(
+            Expressions.parentheses(value1.toExpression()),
+            Expressions.parentheses(Expressions.value(value2)),
+        )
+    }
+
+    /**
+     * Creates an expression that represents the mod of values.
+     * The values are each enclosed in parentheses.
+     *
+     * This is the same as ```(value1) % (value2)```.
+     */
+    @JvmName("modWithParentheses")
+    @SinceJdsl("3.3.2")
+    fun <T : Number, S : T> mod(value1: Expressionable<@Exact T>, value2: Expressionable<S>): Expression<T> {
```

Comment:
In the [Jakarta specification](https://jakarta.ee/specifications/persistence/3.1/jakarta-persistence-spec-3.1.html#a5284), the `MOD` can only accept an integer. Please modify it to accept only integers and not numbers.